### PR TITLE
ref: store last_received datetime as str for issue ingest

### PR DESCRIPTION
**Pull request by asottile-sentry**
in an effort to remove pickling -- this needs to be json serializable and round-trippable

### CODE REVIEW

**Comment by asottile-sentry** [comment]
@getsentry/issue-platform

File: src/sentry/issues/ingest.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def _create_issue_kwargs(
         "first_release": release,
         "data": materialize_metadata(occurrence, event),
     }
-    kwargs["data"]["last_received"] = event.datetime
+    kwargs["data"]["last_received"] = json.datetime_to_str(event.datetime)
```
Comment (wedamija) [inline src/sentry/issues/ingest.py:116]: Could we convert this to a timestamp instead?

Comment (asottile-sentry) [reply src/sentry/issues/ingest.py:116]: actually I would prefer `str` here as it means no special processing is needed to migrate it from pickle -- just loading and storing

Comment (wedamija) [reply src/sentry/issues/ingest.py:116]: Sorry, should have explained my reasoning.

This field is also stored on error issues as a float: https://github.com/getsentry/sentry/blob/0db44d6460577e5a9ae25ed59bae630f218089bf/src/sentry/event_manager.py#L2333. We missed that due to an oversight, but it makes sense to me to keep the format consistent.

But if it's a huge pain to handle the special processing I'm ok with str.

**Review by wedamija — APPROVED** [review]

**Pull request merged**